File: chapter6/sentimental/caesar.py
import cs50
import sys

key = int(sys.argv[1])

if (len(sys.argv) != 2):
    print("Use the proper number of arguments")

ans = 0
var1 = 0
calc = 0
i = 0

source = cs50.get_string("plaintext: ")

print("ciphertext: ", end="")

count = len(source)

while count > 0:
    if (str.isalpha(source[i]) == False):
        print("{}".format(str(source[i])), end="")

    if (str.isupper(source[i]) == True):
        var1 = (ord(source[i]) - 65)
        calc = (var1 + key) % 26
        ans = calc + 65
        print("{}".format(chr(ans)), end="")


    if (str.islower(source[i]) == True):
        var1 = (ord(source[i]) - 97)
        calc = (var1 + key) % 26
        ans = calc + 97
        print("{}".format(chr(ans)), end="")
    i += 1
    count -= 1
print("")

File: chapter2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Use the proper number of arguments");
        return 1;
    }
    int key = atoi(argv[1]);
    int ans = 0, var1 = 0, calc = 0;

    string source = get_string("plaintext:");
    printf("ciphertext: ");
    //take ascii subtract 65, then do equation, then add 65
    for (int i = 0, len = strlen(source); i < len; i++)
    {
        if (isupper(source[i]))
        {
            var1 = (source[i] - 65);
            calc = (var1 + key) % 26;
            ans = calc + 65;
            printf("%c", ans);
        }
        if (islower(source[i]))
        {
            var1 = (source[i] - 97);
            calc = (var1 + key) % 26;
            ans = calc + 97;
            printf("%c", ans);
        }


    }
    printf("\n");
}